### PR TITLE
Validate lengths of fields submitted from addresse form

### PR DESCRIPTION
**Pull request by PululuK**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Some fields in the edit or the create address form need more validations.Validate lengths of fields submitted from addresse form for example !
| Type?         | refacto
| Category?     | CO 
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixe part of https://github.com/PrestaShop/PrestaShop/issues/18284. (I can create a new issue if necessary)
| How to test?  | See description below :

## How to reproduce :  

- Go to FO customer adresses
- Add or edit an address
-  Follow these actions : https://drive.google.com/file/d/1EKSAOBHllfPE_bq6OC9-NNwaQOpJqWxS/view

### > Before 

error500

### > After 

valid-form

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19449)

### CODE REVIEW

File: classes/form/AbstractForm.php [hunk 1]
```diff
@@ -204,4 +212,27 @@ public function setValue($field_name, $value)
 
         return $this;
     }
+
+    /**
```
Comment (Progi1984) [inline classes/form/AbstractForm.php:228]: This method doesn't respect the single responsability principle. It returns a check or a message. Could you fix it, please ?

**Review by Progi1984 — COMMENTED** [review]
Ping @PululuK

File: classes/form/AbstractForm.php (hunk 1 above)
Comment (Progi1984) [reply classes/form/AbstractForm.php:228]: Thanks @PululuK

**Review by Progi1984 — DISMISSED** [review]

File: classes/form/AbstractForm.php (hunk 1 above)
Comment (PululuK) [reply classes/form/AbstractForm.php:228]: Shame on me 😪

Comment (Progi1984) [reply classes/form/AbstractForm.php:228]: Why ?

Comment (PululuK) [reply classes/form/AbstractForm.php:228]: Don't worry ... It's a private joke.
You are too young to understand this 😅

**Comment by PululuK** [comment]
TravisCI is broken 🆘

File: classes/form/AbstractForm.php (hunk 1 above)
Comment (Progi1984) [reply classes/form/AbstractForm.php:228]: Are you sure ? ;)

Comment (PululuK) [reply classes/form/AbstractForm.php:228]: Certainty is uncertain @Progi1984

**Comment by PierreRambaud** [comment]
Yes @PululuK you need to fix the method signature:
```
Fatal error: Uncaught Symfony\Component\Debug\Exception\ContextErrorException: Warning: Declaration of CustomerFormCore::validateFieldLength($fieldName, $maximumLength, $violationMessage) should be compatible with AbstractFormCore::validateFieldLength($field) in /home/travis/build/PrestaShop/PrestaShop/classes/form/CustomerForm.php:267
```

File: classes/form/AbstractForm.php (hunk 1 above)
Comment (Progi1984) [reply classes/form/AbstractForm.php:228]: :smile:

**Review by LouiseBonnard — APPROVED** [review]

**Review by Junebyun — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @PululuK

**Pull request merged**